lib/block-supports/index.php

```diff
@@ -104,12 +104,15 @@ function gutenberg_apply_block_supports( $block_content, $block ) {
 		$block_root->setAttribute( 'style', implode( '; ', $new_styles ) . ';' );
 	}
 
-	$result = '';
-	// phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
-	foreach ( $body_element->childNodes as $child_node ) {
-		$result .= $dom->saveHtml( $child_node );
-	}
-	return $result;
+	// Avoid using `$dom->saveHtml( $node )` because the node results may not produce consistent
+	// whitespace for PHP < 7.3. Saving the root HTML `$dom->saveHtml()` prevents this behavior.
+	$full_html = $dom->saveHtml();
+
+	// Find the <body> open/close tags. The open tag needs to be adjusted so we get inside the tag
+	// and not the tag itself.
+	$start = mb_strpos( $full_html, '<body>', 0, 'UTF-8' ) + mb_strlen( '<body>', 'UTF-8' );
+	$end   = mb_strpos( $full_html, '</body>', $start, 'UTF-8' );
+	return mb_substr( $full_html, $start, $end - $start, 'UTF-8' );
```

Comment:
Since `<body>` and `</body>` do not contain any multi-byte characters, to me it seems somewhat overkill to use the `mb_*` functions. Also, the `mbstring` extension is not always available on PHP installs. 

So I think it would be better to use `strpos()`, `strlen()`, and `substr()`.

```suggestion
	$start = strpos( $full_html, '<body>' ) + strlen( '<body>' );
	$end   = strpos( $full_html, '</body>', $start );
	return substr( $full_html, $start, $end - $start );
```

Reply:
Lastly, should the return value not be `trim()`'ed? Isn't that the original thing that was being fixed with all this?